client/my-sites/term-tree-selector/terms.jsx

```diff
@@ -0,0 +1,407 @@
+				type={ inputType }
+				value={ itemId }
+				onChange={ onChange }
+				disabled={ disabled }
+				checked={ checked }
+			/>
+		);
+
+		return (
+			<div
+				key={ itemId }
+				ref={ setItemRef }
+				className="term-tree-selector__list-item">
+				<label>
+					{ input }
+					<span className="term-tree-selector__label">{ name }</span>
+				</label>
+				{ item.items && (
+					<div className="term-tree-selector__nested-list">
+						{ item.items.map( this.renderItem ) }
+					</div>
+				) }
+			</div>
+		);
+	},
+
+	renderNoResults() {
+		if ( this.hasNoSearchResults() || this.hasNoTerms() ) {
+			return (
+				<div key="no-results" className="term-tree-selector__list-item is-empty">
+					{ ( this.hasNoSearchResults() || ! this.props.emptyMessage ) && (
+						<NoResults
+							createLink={ this.props.createLink } />
+					) }
+					{ this.hasNoTerms() && this.props.emptyMessage }
+				</div>
+			);
+		}
+	},
+
+	renderRow( { index } ) {
+		const item = this.getItem( index );
+		if ( item ) {
+			return this.renderItem( item );
+		}
+
+		return (
+			<div key="placeholder" className="term-tree-selector__list-item is-placeholder">
+				<label>
+					<input
+						type={ this.props.multiple ? 'checkbox' : 'radio' }
+						disabled
+						className="term-tree-selector__input" />
+					<span className="term-tree-selector__label">
+						{ this.props.translate( 'Loading…' ) }
+					</span>
+				</label>
+			</div>
+		);
+	},
+
+	render() {
+		const rowCount = this.getRowCount();
+		const isCompact = this.isCompact();
+		const searchLength = this.state.searchTerm.length;
+		const showSearch = ( searchLength > 0 || ! isCompact ) &&
+			( this.props.terms || ( ! this.props.terms && searchLength > 0 ) );
+		const { className, loading, siteId, taxonomy, query } = this.props;
+		const classes = classNames( 'term-tree-selector', className, {
+			'is-loading': loading,
+			'is-compact': isCompact
+		} );
+
+		return (
+			<div ref={ this.setSelectorRef } className={ classes }>
+				{ this.state.requestedPages.map( ( page ) => (
+					<QueryTerms
+						key={ `query-${ page }` }
+						siteId={ siteId }
+						taxonomy={ taxonomy }
+						query={ { ...query, page } } />
+				) ) }
+				{ showSearch && (
+					<Search
+						searchTerm={ this.state.searchTerm }
+						onSearch={ this.onSearch } />
+				) }
+				<VirtualScroll
+					ref={ this.setVirtualScrollRef }
+					width={ this.getResultsWidth() }
+					height={ isCompact ? this.getCompactContainerHeight() : 300 }
+					onRowsRendered={ this.setRequestedPages }
+					rowCount={ rowCount }
+					estimatedRowSize={ ITEM_HEIGHT }
+					rowHeight={ this.getRowHeight }
+					rowRenderer={ this.renderRow }
+					noRowsRenderer={ this.renderNoResults }
+					className="term-tree-selector__results" />
+			</div>
+		);
+	}
+} );
+
+export default connect( ( state, ownProps ) => {
+	const siteId = getSelectedSiteId( state );
+	const { taxonomy, query } = ownProps;
+	query.number = query.number || 100;
```

Comment:
>changing to:

Nice catch! Looks like the right solution. I'd be fine with including it in these changes. Wonder if it'd not be too much more work to try to write a (currently-failing) test to reflect the need for that specific change.